tools/cellprofiler/common.xml

```diff
@@ -0,0 +1,395 @@
+              </when>
+            </conditional>       
+    </xml>
+    <xml name="nameTypeRuleMatchingFile">
+      <param name="operator" type="select">
+          <option value="does">Does</option>
+          <option value="doesnot">Doesnot</option>
+      </param>
+      <param name="contain" type="select">
+          <option value="contain">Contain</option>
+          <option value="Contain regular expression">Contain regular expression</option>
+          <option value="startwith">Start with</option>
+          <option value="endwith">End with</option>
+          <option value="Exactly match">Exactly match</option>
+      </param>   
+      <param name="match_value" type="text"/>   
+    </xml>
+    <xml name="imgMatchingRules">
+        <repeat name="r_match_rule" title="Rules">
+        <param name="match_all_any" type="select" display="radio" label="Match the following rules">
+            <option value="and">All</option>
+            <option value="or">Any</option>
+        </param>          
+          <conditional name="con_match">
+              <param name="rule_type" type="select" label="Select rule criteria">
+                  <option value="file">File</option>
+                  <option value="directory">Directory</option>
+                  <option value="extension">Extension</option>
+                  <option value="image">Image</option>
+                  <option value="metadata">Metadata</option>
+              </param>
+              <when value="file">
+                  <expand macro="nameTypeRuleMatchingFile"/>
+            </when>  
+            <when value="Directory">
+                  <expand macro="nameTypeRuleMatchingFile"/>
+            </when>  
+          </conditional>
+          <param name="name_to_assign" label="Name to assign these images" type="text" />
+          <conditional name="con_select_image_type">
+                <param name="select_image_type" type="select" label="Select the image type">
+                  <option value="Grayscale image">Grayscale image</option>
+                  <option value="Color image">Color image</option>
+                  <option value="Binary mask">Binary mask</option>
+                </param>         
+                 <when value ="Grayscale image">
+                    <expand macro="imgTypeCondition" />
+                </when> 
+                <when value="Color image">
+                    <expand macro="imgTypeCondition" />
+                </when>
+            </conditional>          
+        </repeat>
+
+
+    </xml>
+  </macros>
+  <configfiles>
+    <inputs name="inputs" filename="input.json" />
+    <configfile name="script_file">
+import json
+import sys
+import os
+
+FOURSPACES="    "
+NEWLINE="\n"
+
+input_json_path = sys.argv[1]
+
+
+params = json.load(open(input_json_path, "r"))
+
+#params_out = json.dumps(params, indent=4, sort_keys=True)
+
+def write_images():
+  filter_images = params['images']['filter_images']
+
+  f.write(NEWLINE)

```

Comment:
Please try to assemble this as a string and only write once in the end.

Reply:
do we really need to change this....
can do if really needed, just the matter of time to change these across all files and re-test

Reply:
No, not strictly needed, but this can be a performance hit, a small ..